### PR TITLE
Decompile unk_02002C14.s into font.c

### PR DESCRIPTION
**Pull request by remicalixte**
thought it would be an easy file...
I left a NONMATCHING for `FUN_02002CF8` because I couldn't reproduce what happens with the index: `param0` is used to index a array of u32. For some reason it's shifted left by 2 several times in the original assembly whereas it's only computed once in my attempts.

### CODE REVIEW

File: arm9/src/font.c [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+THUMB_FUNC void FUN_02002CC0(s32 param0)
+{
+    GF_ASSERT(param0 < 4);
+    GF_ASSERT(UNK_02106FC8->unk94[param0] != NULL);
+
+    FUN_020215E0(UNK_02106FC8->unk94[param0], 1, 0);
+}
+
+#ifdef NONMATCHING
+THUMB_FUNC void FUN_02002CF8(s32 param0)
```
Comment (PikalaxALT) [inline arm9/src/font.c:64]: The following appears to match. Note: parameter type is important, mwccarm treats `s32` and `int` differently sometimes.

```c
void FUN_02002CF8(int a0)
{
    s32 r1;

    GF_ASSERT(a0 < 4);

    if (UNK_02106FC8->unk84[a0] != NULL)
    {
        for (r1 = 0; r1 < 4u; r1++)
        {
            if (r1 != a0 
             && UNK_020ECB54[r1][0] == UNK_020ECB54[a0][0]
             && UNK_02106FC8->unk94[r1] != NULL)
            {
                UNK_02106FC8->unk84[r1] = UNK_02106FC8->unk84[a0];
                break;
            }
        }
        if (r1 == 4u)
        {
            FreeToHeap(UNK_02106FC8->unk84[a0]);
            UNK_02106FC8->unk84[a0] = NULL;
        }
    }
    if (UNK_02106FC8->unk94[a0] != NULL)
    {
        FUN_020215C8(UNK_02106FC8->unk94[a0]);
        UNK_02106FC8->unk94[a0] = NULL;
    }
}
```

File: arm9/src/font.c [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+            ret = gFontInfos[fontId].shadowColor;
+            break;
+    }
+
+    return ret;
+}
+
+THUMB_FUNC void FUN_02002ED0(u32 layer, u32 baseAddr, u32 heap_id)
+{
+    GfGfxLoader_GXLoadPal(14, 6, layer, baseAddr, 0x20, heap_id);
```
Comment (PikalaxALT) [inline arm9/src/font.c:265]: Use the appropriate enum from filesystem.h, and include the corresponding .naix which defines the NARC member enum

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: arm9/src/font.c (hunk 1 above)
Comment (remicalixte) [reply arm9/src/font.c:64]: wow thanks. It seems both using an `int` and turning `UNK_020ECB54` into a 2d array was necessary.

File: arm9/src/font.c (hunk 2 above)
Comment (remicalixte) [reply arm9/src/font.c]: It's used to load assets right? That's why I couldn't find an equivalent in pokeemerald

**Review by PikalaxALT — APPROVED** [review]

**Review by red031000 — APPROVED** [review]

**Pull request merged**